### DISCUSSION TITLE
stall detection on a complicated drive train

### DISCUSSION THREAD

**Discussion by BertLindeman** [pybricks/support#1178] [Q&A]
Hello,

Trying to program [rebrickable MOC-129823](https://rebrickable.com/mocs/MOC-129823/3Lego1Math/robotic-arm-tictactoe/).
The gripper is a long way away from the gripper motor with many gears between them.

I am trying to get the gripper calibrated, so I know at which motor angle it is closed / open.

But all I get is rattling gears. 
I tried to fiddle with ```gripper.control.stall_tolerances(100, 10)```
But it seems not to help.

Is there a way to get this done?

Thank you,

Bert

[EDIT] running: ```('primehub', '3.3.0b8', 'v1.20.0-23-g6c633a8dd on 2023-07-07')```

**Comment by laurensvalk** [comment]
How about using the `duty_limit` on `run_until_stalled`?

Or reduce the available `torque` in `control.limits`.

**Reply by BertLindeman** [reply]
Will try and report back.
Thank you Laurens

**Comment by BertLindeman** [comment]
YES 😄 
torque is **the** medicine for this gripper.
Default was 263 (device dependend?) and
```gripper.control.limits(torque=50)```
makes the motor neatly stop before the rattling gears.
Thanks again Laurens
Now I can get on.

Bert
